data/trie/trieStorageManager_test.go

```diff
@@ -0,0 +1,379 @@
+	t.Parallel()
+
+	ts, err := NewTrieStorageManager(mock.NewMemDbMock(), config.DBConfig{}, nil)
+	assert.Nil(t, ts)
+	assert.Equal(t, ErrNilEvictionWaitingList, err)
+}
+
+func TestNewTrieStorageManagerOkVals(t *testing.T) {
+	t.Parallel()
+
+	ts, err := NewTrieStorageManager(mock.NewMemDbMock(), config.DBConfig{}, &mock.EvictionWaitingList{})
+	assert.Nil(t, err)
+	assert.NotNil(t, ts)
+}
+
+func TestTrieDatabasePruning(t *testing.T) {
+	t.Parallel()
+
+	db := mock.NewMemDbMock()
+	msh, hsh := getTestMarshAndHasher()
+	size := 1
+	evictionWaitList, _ := mock.NewEvictionWaitingList(size, mock.NewMemDbMock(), msh)
+	trieStorage, _ := NewTrieStorageManager(db, config.DBConfig{}, evictionWaitList)
+
+	tr := &patriciaMerkleTrie{
+		trieStorage: trieStorage,
+		oldHashes:   make([][]byte, 0),
+		oldRoot:     make([]byte, 0),
+		marshalizer: msh,
+		hasher:      hsh,
+	}
+
+	_ = tr.Update([]byte("doe"), []byte("reindeer"))
+	_ = tr.Update([]byte("dog"), []byte("puppy"))
+	_ = tr.Update([]byte("dogglesworth"), []byte("cat"))
+	_ = tr.Commit()
+
+	key := []byte{6, 4, 6, 15, 6, 7, 16}
+	oldHashes := make([][]byte, 0)
+	n := tr.root
+	rootHash, _ := tr.Root()
+	oldHashes = append(oldHashes, rootHash)
+
+	for i := 0; i < 3; i++ {
+		n, key, _ = n.getNext(key, db)
+		oldHashes = append(oldHashes, n.getHash())
+	}
+
+	_ = tr.Update([]byte("dog"), []byte("doee"))
+	_ = tr.Commit()
+
+	err := tr.Prune(rootHash, data.OldRoot)
+	assert.Nil(t, err)
+
+	for i := range oldHashes {
+		encNode, err := tr.Database().Get(oldHashes[i])
+		assert.Nil(t, encNode)
+		assert.NotNil(t, err)
+	}
+}
+
+func TestRecreateTrieFromSnapshotDb(t *testing.T) {
+	t.Parallel()
+
+	tr, trieStorage, _ := newEmptyTrie()
+	_ = tr.Update([]byte("doe"), []byte("reindeer"))
+	_ = tr.Update([]byte("dog"), []byte("puppy"))
+	_ = tr.Update([]byte("dogglesworth"), []byte("cat"))
+
+	_ = tr.Commit()
+	rootHash, _ := tr.Root()
+	_ = tr.Snapshot()
+
+	for trieStorage.snapshotsBuffer.len() != 0 {
+		time.Sleep(snapshotDelay)
+	}
+
+	_ = tr.Update([]byte("doge"), []byte("doge"))
+	_ = tr.Commit()
+	_ = tr.Prune(rootHash, data.OldRoot)
+
+	val, err := tr.Database().Get(rootHash)
+	assert.Nil(t, val)
+	assert.NotNil(t, err)
+
+	newTrie, err := tr.Recreate(rootHash)
+	assert.Nil(t, err)
+	assert.NotNil(t, newTrie)
+}
+
+func TestEachSnapshotCreatesOwnDatabase(t *testing.T) {
+	t.Parallel()
+
+	testVals := []struct {
+		key   []byte
+		value []byte
+	}{
+		{[]byte("doe"), []byte("reindeer")},
+		{[]byte("dog"), []byte("puppy")},
+		{[]byte("dogglesworth"), []byte("cat")},
+	}
+
+	tr, trieStorage, _ := newEmptyTrie()
+
+	for _, testVal := range testVals {
+		_ = tr.Update(testVal.key, testVal.value)
+		_ = tr.Commit()
+		_ = tr.Snapshot()
+		for trieStorage.snapshotsBuffer.len() != 0 {
+			time.Sleep(snapshotDelay)
+		}
+
+		snapshotId := strconv.Itoa(trieStorage.snapshotId - 1)
+		snapshotPath := path.Join(trieStorage.snapshotDbCfg.FilePath, snapshotId)
+		f, _ := os.Stat(snapshotPath)
```

Comment:
`if f, err := os.Stat(snapshotPath); !os.IsNotExist(err) {`

Reply:
I don't check the err here.